src/main/webapp/assets/script.js

```diff
@@ -72,12 +73,89 @@ function onLoad() {
   collegeLocation = fetchFakeCollegeLocation();
 
   // Add the embedded map to the page.
-  const script = document.createElement('script');
-  script.src = `https://maps.googleapis.com/maps/api/js?key=AIzaSyB_iMemNeUtO8D7kGTmagTgluRlsiFTcDA&callback=initMap`;
-  script.defer = true;
-  script.async = true;
-  window.initMap = initMap;
-  document.head.appendChild(script);
+  getSecretFor('javascript-maps-api').then((key) => {
```

Comment:
What happens when null is returned e.g. the try block for `getSecretFor` fails?

Reply:
Sounds good!